src/App.tsx

```diff
@@ -1,9 +1,28 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-export const App: React.FC = () => {
+export const App: FC = () => {
+  const [visibleMovies, setVisibleMovies] = useState(moviesFromServer);
+  const [query, setQuery] = useState('');
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  useEffect(() => {
+    const filteredMovies = moviesFromServer.filter((movie) => {
+      const lowerCaseSearchQuery = query.toLowerCase().trim();
+
+      return [movie.title, movie.description].some(
+        (text) => text.toLowerCase().includes(lowerCaseSearchQuery),
+      );
+    });
+
+    setVisibleMovies(filteredMovies);
+  }, [query]);
```

Comment:
This code look nice but let's simplify it
```suggestion
    const lowerCaseSearchQuery = query.toLowerCase().trim();
    
    const visibleMovies = moviesFromServer.filter(movie => {
      return [movie.title, movie.description].some(
        (text) => text.toLowerCase().includes(lowerCaseSearchQuery),
      );
  })
```